core/templates/expressions/expression-syntax-tree.service.ts

```diff
@@ -16,417 +16,447 @@
-      // minification (when running the deployment scripts).
-      /* eslint-disable quote-props */
-      'if': {
-        eval: function(args) {
-          // TODO(kashida): Make this short-circuit.
-          verifyNumArgs(args, 3);
-          return args[0] ? args[1] : args[2];
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 3);
-          _verifyArgTypesMatchExpectedType(
-            [args[0]], PARAMETER_TYPES.UNICODE_STRING);
-          _verifyArgTypesMatch(args[1], args[2]);
-          return args[1];
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 2);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.UNICODE_STRING);
+        return AppConstants.PARAMETER_TYPES.UNICODE_STRING;
+      }
+    },
+    // Note that removing quotation marks from this key causes issues with
+    // minification (when running the deployment scripts).
+    /* eslint-disable quote-props */
+    if: {
+      eval: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 3);
+        return args[0] ? args[1] : args[2];
       },
-      'floor': {
-        eval: function(args) {
-          verifyNumArgs(args, 1);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          return Math.floor(numericArgs[0]);
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 1);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.REAL;
-        }
+      getType: (args: Array<string>): string => {
+        this.verifyNumArgs(args, 3);
+        this._verifyArgTypesMatchExpectedType([args[0]],
+          AppConstants.PARAMETER_TYPES.UNICODE_STRING);
+        this._verifyArgTypesMatch(args[1], args[2]);
+        return args[1];
+      }
+    },
+    floor: {
+      eval: (args: Array<number|string>): number => {
+        this.verifyNumArgs(args, 1);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        return Math.floor(numericArgs[0]);
       },
-      'pow': {
-        eval: function(args) {
-          verifyNumArgs(args, 2);
-          var numericArgs = _coerceAllArgsToNumber(args);
-          return Math.pow(args[0], args[1]);
-        },
-        getType: function(args) {
-          verifyNumArgs(args, 2);
-          _verifyArgTypesMatchExpectedType(args, PARAMETER_TYPES.REAL);
-          return PARAMETER_TYPES.REAL;
-        }
+      getType: (args: Array<number|string>): string => {
+        this.verifyNumArgs(args, 1);
+        this._verifyArgTypesMatchExpectedType(args,
+          AppConstants.PARAMETER_TYPES.REAL);
+        return AppConstants.PARAMETER_TYPES.REAL;
+      }
+    },
+    pow: {
+      eval: (args: Array<number|string>): number => {
+        this.verifyNumArgs(args, 2);
+        var numericArgs = this._coerceAllArgsToNumber(args);
+        // NOTE TO DEVELOPERS: we need to coerce args items with
```

Comment:
Please change this comment as well. Clearly mentioning why we need + here. It's okay if the comment becomes too long, it should be clear to anyone who reads it even after years :).

Reply:
@bansalnitish I was rewriting the comment until one variable caught my eyes 🤭
`pow: eval: var numericArgs = this._coerceAllArgsToNumber(args)` is indicating me in my editor that it's unused. I mean gosh the whole time I was debugging and I should've seen this in the first place! So embarrassing 😓

Result:
```
eval: (args: Array<string>): number => {
        this.verifyNumArgs(args, 2);
        var numericArgs = this._coerceAllArgsToNumber(args);
        return Math.pow(numericArgs[0], numericArgs[1]);
      },
```

Reply:
No worries, I'm glad you got it :)